openapi-style-guide.md

```diff
@@ -0,0 +1,208 @@
+- The operation should have the `x-ms-pageable` annotation
+- The operation response should contain a top-level `value` property of type array and required
+- The operation response should contain a top-level `nextLink` property of type string and optional
+- If the operation has a `skip` parameter, it must be an integer and optional
+- If the operation has a `top` parameter, it must be an integer, optional, and have a documented default and maximum value
+- If the operation has a `maxpagesize` parameter, it must be an integer, optional, and have a documented default and maximum value
+
+### Error response
+
+All operations should have a "default" (error) response with a response body that conforms to the Azure API Guidelines.
+
+All `4xx` and `5xx` responses should specify `x-ms-error-response: true` except for `404` response of HEAD operation.
+
+Every error response should include the `x-ms-error-code` response header in the `headers` of the response.
+
+Example:
+```
+      "400": {
+        "description": "Bad Request",
+        "headers": {
+          "x-ms-error-code": {
+            "type": "string",
+            "description": "The error code."
+          }
+        },
+        "schema": {
+          "$ref": "#/definitions/ErrorResponse"
+        },
+        "x-ms-error-response": true
+      },
+```
+
+### Response headers
+
+A 202 response should include the `Operation-Location` response header in the `headers` of the response.
+
+Example:
+```
+    "responses": {
+      "202": {
+        "description": "The service has accepted the request and will start processing later.",
+        "headers": {
+          "Operation-Location": {
+            "description": "URL to query for status of the operation.",
+            "type": "string"
+          }
+        }
+      },
+```
+
+<!-- --------------------------------------------------------------- -->
+
+## Parameters
+
+### Parameter names
+
+All parameter names for an operation -- including parameters defined at the path level -- should be case-insensitive unique.
+
+### Descriptions
+
+Every parameter should have a description.
+
+### Format
+
+Format must be one of the values [defined by OpenAPI][openapi-data-types] and recognized by the Azure tooling.
+
+Integer properties must specify a format of either `int32` or `int64`.
+
+### Default
+
+Required parameters should not specify a default value.
+
+<!-- --------------------------------------------------------------- -->
+
+## Schemas
+
+OpenAPI uses JSON Schema as the means for describing request and response body payloads.
+
+### Schema names
+
+Schema names should be [Pascal case][wikipedia-camel-case].
+
+### Descriptions
+
+Every schema should have a description or a title.
+
+Every schema property should have a description.
+
+### Format
+
+Integer properties must specify a format of either `int32` and `int64`.
+
+Format must be one of the values [defined by OpenAPI][openapi-data-types] and recognized by the Azure tooling.
```

Comment:
This looks like it's duplicated?

Reply:
It does look that way but that's because the same guidance applies to _parameters_ (above) and _properties_ (here).

Reply:
Do we need both?  Should we just have a section on data types that we link to from params/properties?  I'm fine with this now though and we can wait until the doc becomes unwieldy before we start abstracting sections.